server/neptune/workflows/activities/terraform.go

```diff
@@ -98,10 +99,33 @@ func NewTerraformActivities(
 	}
 }
 
+func getEnvs(directEnvs map[string]string, dynamicEnvs []EnvVar) (map[string]string, error) {
+	envs := make(map[string]string)
+
+	for k, v := range directEnvs {
+		envs[k] = v
+	}
+
+	for _, e := range dynamicEnvs {
+		v, err := e.GetValue()
+
+		if err != nil {
+			return envs, errors.Wrap(err, fmt.Sprintf("loading dynamic env var with name %s", e.Name))
+		}
+
+		envs[e.Name] = v
+	}
+
+	return envs, nil
+}
+
 // Terraform Init
 type TerraformInitRequest struct {
-	Args                 []terraform.Argument
+	Args []terraform.Argument
+	// deprecated: Use DynamicEnvs instead
+	// remove once we are sure this isn't used
 	Envs                 map[string]string
```

Comment:
This apart of your attempt to version? Why not just remove otherwise?

Reply:
I think we need this because in progress workflows that have already passed the version check in history might execute these activities and therefore the old field would be serialized.